svgpathtools/path.py

```diff
@@ -274,9 +273,28 @@ def to_complex(v):
     elif isinstance(curve, Arc):
         new_start = to_complex(tf.dot(to_point(curve.start)))
         new_end = to_complex(tf.dot(to_point(curve.end)))
-        new_radius = to_complex(tf.dot(to_vector(curve.radius)))
-        return Arc(new_start, radius=new_radius, rotation=curve.rotation,
-                   large_arc=curve.large_arc, sweep=curve.sweep, end=new_end)
+        
+        # Based on https://math.stackexchange.com/questions/2349726/compute-the-major-and-minor-axis-of-an-ellipse-after-linearly-transforming-it
+        rx2 = curve.radius.real ** 2
+        ry2 = curve.radius.imag ** 2
+
+        Q = np.array([[1/rx2, 0], [0, 1/ry2]])
+        invT = np.linalg.inv(tf[:2,:2])
+        D = invT.T @ Q @ invT
+
+        eigvals = np.linalg.eigvals(D)
+
+        rx = 1 / np.sqrt(eigvals[0])
+        ry = 1 / np.sqrt(eigvals[1])
+
+        new_radius = complex(rx, ry)
+
+        if new_radius.real == 0 or new_radius.imag == 0 :
+            return Line(new_start, new_end)
+        else : 
+            return Arc(new_start, radius=new_radius, rotation=curve.rotation,
+                       large_arc=curve.large_arc, sweep=curve.sweep, end=new_end)
```

Comment:
```suggestion
                       large_arc=curve.large_arc, sweep=curve.sweep, end=new_end,
                       autoscale_radius=False)
```
I believe it would be good to disable autoscaling. If the transform is applied correctly, no scaling should be needed. If not, better to raise an error about a failed transformation than silently trying to patch up the result?